### PR TITLE
Fixes #11794: quality assistant should merge approved PR

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/11794

### CODE REVIEW

**Comment by peckpeck** [comment]
Commit modified

File: scripts/rudder-dev/quality-assistant [hunk 1]
```diff
@@ -124,6 +125,48 @@ def repo_merge(repo, alt_dir):
...
+  # List approved PR
+  api_url = "https://api.github.com/repos/Normation/{repo}/pulls?state=open"
+  url = api_url.format(repo=repo)
+  data = github_call(url)
+  for pr_info in data:
+    # list reviews of this PR (always in chronological order)
+    api_url = "https://api.github.com/repos/Normation/{repo}/pulls/{id}/reviews"
+    url = api_url.format(repo=repo, id=data["number"])
+    status = {}
+    for review in github_call(url):
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/quality-assistant]: I would reorganize:

`
if freeze:
  skip (freeze) 
else:
  reviews = github_call(url)
  if len(reviews) == 0:
    skip (no review)
  else:
    for review in reviews:
      status[review['user']['login']] = review['state']
    if "CHANGE_REQUESTED" in status.values:
      reassign
    else:
      merge
`

Comment (VinceMacBuche) [reply scripts/rudder-dev/quality-assistant]: This has also a side effect of reassing at every run (but don't know if it's a problem though)

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**